galter_subjects_utils/cli.py

```diff
@@ -79,3 +81,33 @@ def lcsh(**parameters):
     filepath = write_jsonl(converter, parameters["output_file"])
 
     print(f"LCSH terms written here {filepath}")
+
+
+keep_trace_field_help = "Dotted field path to where trace should be kept."
+keep_trace_tmpl_help = "Template with expandable '{subject}' to be saved."
+
+
+@main.command("update")
+@click.argument(
+    "deltas-file",
+    type=click.Path(path_type=Path, exists=True, dir_okay=False),
+)
+@click.option(
+    "--output-file", "-o",
+    type=click.Path(path_type=Path),
+    default=defaults["output-file"] / "updated_records.csv",
+)
+@click.option("--keep-trace-field", "-f", help=keep_trace_field_help)
+@click.option("--keep-trace-template", "-t", help=keep_trace_tmpl_help)
+@with_appcontext
+def update_subjects(**parameters):
+    """Update subjects in running instance according to deltas file."""
+    deltas = [d for d in read_csv(parameters["deltas_file"])]
+    logger = SubjectDeltaLogger(filepath=parameters["output_file"])
+    updater = SubjectDeltaUpdater(deltas, logger)
+    keep_trace = KeepTrace(
+        field=parameters.get("keep_trace_field") or None,
+        template=parameters.get("keep_trace_template") or None
+    )
+    updater.update(keep_trace)
```

Comment:
TBH `keep_trace` could be passed as a constructor argument. For these big one pass algorithms, OO is very weird. Thoughts on this welcome!

Reply:
In a way I put it there to elicit some response discussion ;). For these kinds of algorithms, I am having a hard time seeing why methods should take an argument when it really operates as loading up the constructor and firing it off with a method call. But I don't know that I have a solid argument against it either... As you say, keeping with the approach so far is a good enough argument.  I will put it back in the constructor.

I was thinking a lot about OO structure here and I think the bottom line for this kind of algorithm is that it's easier to test / there no globally accessible state. Other OO stuff doesn't really matter here.